packages/react-spinbutton/src/components/SpinButton/SpinButton.types.ts

```diff
@@ -1,20 +1,163 @@
+  /**
+   * The root element of SpinButton is a container `<div>`.
+   * The root slot receives the `className` and `style` specified on the `<SpinButton>`.
+   * All other native props are applied to the primary slot: `input`.
+   */
   root: IntrinsicShorthandProps<'div'>;
+
+  /**
+   * Input that displays the current value and accepts direct input from the user.
+   * Displayed value is formatted.
+   *
+   * This is the primary slot.
+   */
+  input: IntrinsicShorthandProps<'input'>;
+
+  /**
+   * Renders the increment button.
+   */
+  incrementButton: IntrinsicShorthandProps<'button'>;
+
+  /**
+   * Renders the decrement button.
+   */
+  decrementButton: IntrinsicShorthandProps<'button'>;
+
+  /**
+   * Renders the label.
+   */
+  label: ObjectShorthandProps<LabelProps> | IntrinsicShorthandProps<'label'>;
+};
+
+export type SpinButtonChangeData = {
+  /**
+   * New value after the change.
+   * E.g., `1`
+   */
+  value: number;
+
+  /**
+   * New formatted value after the change.
+   * E.g., `"$1.00"`
+   */
+  formattedValue: string;
+
+  /**
+   * Previous value, before the change.
+   */
+  prevValue: number;
+
+  /**
+   * Previous formatted value, before the change.
+   */
+  prevFormattedValue: string;
 };
 
+export type SpinButtonFormatter = (value: number) => string;
+export type SpinButtonParser = (formattedValue: string) => number;
+export type SpinButtonKeyboardDirection = 'none' | 'up' | 'down';
+
 export type SpinButtonCommons = {
-  // TODO Add things shared between props and state here
+  /**
+   * Initial value of the control (assumed to be valid). Updates to this prop will not be respected.
+   *
+   * Use this if you intend for the SpinButton to be an uncontrolled component which maintains its
+   * own value. For a controlled component, use `value` instead. (Mutually exclusive with `value`.)
+   * @defaultvalue 0
+   */
+  defaultValue: number; // Initial value of the
+
+  /**
+   * Current value of the control (assumed to be valid).
+   *
+   * Only provide this if the SpinButton is a controlled component where you are maintaining its
+   * current state and passing updates based on change events; otherwise, use the `defaultValue`
+   * property. (Mutually exclusive with `defaultValue`.)
+   */
+  value: number;
+
+  /**
+   * Min value of the control. If not provided, the control has no minimum value.
+   */
+  min: number;
+
+  /**
+   * Max value of the control. If not provided, the control has no maximum value.
+   */
+  max: number;
+
+  /**
+   * Difference between two adjacent values of the control.
+   * This value is used to calculate the precision of the input if no `precision` is given.
+   * The precision calculated this way will always be \>= 0.
+   * @defaultvalue 1
+   */
+  step: number;
+
+  /**
+   * Custom className for the control.
+   */
+  className: string;
+
+  /**
+   * Descriptive label for the control.
+   */
+  label: string;
+
```

Comment:
```suggestion
```

Currently `label` is already defined as a slot. It should not be included there, correct?